spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilter.java

```diff
@@ -241,6 +241,7 @@ private void addProxyPrefix(RequestContext ctx, Route route) {
 				else {
 					newPrefixBuilder.append(prefix);
 				}
+				newPrefixBuilder.append(",");
```

Comment:
Wouldnt this result in a `x-forwarded-prefix` ending in a `,` if the headers wasnt already present?  I think a better approach would be to add the `,` to the end of the header value if it exists and is not already present in the code above.

Reply:
Nope, the line a little bit further in teh code always add the new prefix behind the added comma

Reply:
Sure if the header already existed in the request, but if it didnt than it would just append it at the end.  

I wouldnt make an assumption that this situation is covered in a test somewhere

Reply:
```
if (StringUtils.hasText(route.getPrefix())) {
  // So the route does have a prefix configured
  StringBuilder newPrefixBuilder = new StringBuilder();
  if (prefix != null) {
    // The prefix is also set
    if (prefix.endsWith("/") && route.getPrefix().startsWith("/")) {
      newPrefixBuilder.append(prefix, 0, prefix.length() - 1);
    }
    else {
      newPrefixBuilder.append(prefix);
    }
    // newPrefixBuilder now contains the prefix AND we know that the route does have a prefix
    newPrefixBuilder.append(",");
  }
  // Add the configures route prefix to the builder. Possibly containing an existing prefix appended with a comma
  newPrefixBuilder.append(route.getPrefix());
  prefix = newPrefixBuilder.toString();
}
```